From 1164426271768f143d2ef6d105f87942346d17c2 Mon Sep 17 00:00:00 2001
From: Michal Simek <monstr@monstr.eu>
Date: Mon, 3 Oct 2011 12:52:47 +0200
Subject: [PATCH] Xilinx: Axi Ethernet Driver: Update to the latest kernel version

- Add missing headers
- Change driver registration to reflect current state

Signed-off-by: Michal Simek <monstr@monstr.eu>
---
 drivers/net/ethernet/xilinx/xilinx_axienet.h      |    1 +
 drivers/net/ethernet/xilinx/xilinx_axienet_main.c |   15 +++++++++++----
 2 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_axienet.h b/drivers/net/ethernet/xilinx/xilinx_axienet.h
index 736ddca..ea50de2 100644
--- a/drivers/net/ethernet/xilinx/xilinx_axienet.h
+++ b/drivers/net/ethernet/xilinx/xilinx_axienet.h
@@ -10,6 +10,7 @@
 
 #include <linux/netdevice.h>
 #include <linux/spinlock.h>
+#include <linux/interrupt.h>
 
 
 /* packet size info */
diff --git a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
index 93152a1..f390823 100644
--- a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
+++ b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
@@ -26,10 +26,17 @@
 #include <linux/of_mdio.h>
 #include <linux/of_platform.h>
 #include <linux/of_address.h>
+#include <linux/of.h>
+#include <linux/of_device.h>
+#include <linux/of_mdio.h>
+#include <linux/of_platform.h>
+#include <linux/of_address.h>
 #include <linux/skbuff.h>
 #include <linux/spinlock.h>
 #include <linux/phy.h>
 #include <linux/mii.h>
+#include <linux/dma-mapping.h>
+#include <linux/io.h>
 
 #include "xilinx_axienet.h"
 
@@ -1806,7 +1813,7 @@ static void axienet_dma_err_handler(unsigned long data)
  * axienet_local. It registers the Ethernet device.
  **/
 static int __init
-axienet_of_probe(struct platform_device *op, const struct of_device_id *match)
+axienet_of_probe(struct platform_device *op)
 {
 	struct device_node *np;
 	struct axienet_local *lp;
@@ -1997,7 +2004,7 @@ static int __devexit axienet_of_remove(struct platform_device *op)
 	return 0;
 }
 
-static struct of_platform_driver axienet_of_driver = {
+static struct platform_driver axienet_of_driver = {
 	.probe = axienet_of_probe,
 	.remove = __devexit_p(axienet_of_remove),
 	.driver = {
@@ -2015,7 +2022,7 @@ static struct of_platform_driver axienet_of_driver = {
  **/
 static int __init axienet_init(void)
 {
-	return of_register_platform_driver(&axienet_of_driver);
+	return platform_driver_register(&axienet_of_driver);
 }
 
 module_init(axienet_init);
@@ -2028,7 +2035,7 @@ module_init(axienet_init);
  **/
 static void __exit axienet_exit(void)
 {
-	of_unregister_platform_driver(&axienet_of_driver);
+	platform_driver_unregister(&axienet_of_driver);
 }
 module_exit(axienet_exit);
 
-- 
1.7.0.4

